### PR TITLE
container/list: introduce x/container and move List[T] there

### PR DESCRIPTION
**Pull request by amery**
this replaces https://github.com/darvaza-proxy/core/pull/84

## Summary by CodeRabbit

- **New Features**
	- Introduced a new copyright notice and licensing terms for the software.
	- Enhanced the `List` type for improved type safety and functionality, including new methods like `Purge`.

- **Bug Fixes**
	- Improved error handling in list retrieval methods to ensure valid elements are returned.

- **Documentation**
	- Updated module declarations and dependencies for better clarity and management.

- **Refactor**
	- Transitioned to using a standardized list implementation across various components, ensuring consistency in list management.

### CODE REVIEW

File: container/list/list.go [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
 type List[T any] list.List
 
 // Sys returns the native [list.List]
-func (l *List[_]) Sys() *list.List {
+func (l *List[T]) Sys() *list.List {
 	if l == nil {
 		return nil
 	}
 	return (*list.List)(l)
 }
```
[the remark this answers is not in the document]
Comment (amery) [reply container/list/list.go:16]: no one likes my manhole :(

**Review by karasz — CHANGES_REQUESTED** [review]
x is already MIT licensed to JPI Ltd, container is MIT licensed to Apptly, this might confuse people

**Comment by amery** [comment]
> x is already MIT licensed to JPI Ltd, container is MIT licensed to Apptly, this might confuse people

`x/container/LICENCE.txt` amended and PR rebased.

@karasz

File: container/LICENCE.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+OTHER DEALINGS IN THE SOFTWARE.
```
[the remark this answers is not in the document]
Comment (amery) [reply container/LICENCE.txt:19]: should code that didn't exist in 2021 have 2021 in their copyright notice?

**Pull request merged**